packages/client/src/components/project/input/Datasets.tsx

```diff
@@ -27,18 +27,22 @@ const useStyles = makeStyles((theme: Theme) =>
   })
 );
 
-export default function Datasets(props: { id: string; selected: GetProjectData_project_datasets[] }): ReactElement {
+export default function Datasets(props: { id: string; selected: GetProjectData_project_dataset | null }): ReactElement {
   const classes = useStyles();
 
   const { data, loading, error } = useQuery<GetDatasets, GetDatasets>(GET_DATASETS);
-  const datasetNames = props.selected.map((dataset) => dataset.name);
+
+  let datasetName = "Dataset";
+  if (props.selected != null) {
+    datasetName = props.selected.name;
+  }
```

Comment:
Try something like this and remove the code above on lines 34-38. Reach out if it doesn't work. I probably have a question mark in the wrong spot.

```suggestion
  const datasetName = props.selected?.name ?? "Dataset"
```

I am also not convinced that we want a default of "dataset". Shouldn't it be null/none?

Reply:
@technoAl is there still a default?

Reply:
it's the empty string "" It works well